Data on Graham
Transcriptome Data:

/home/rajni/projects/def-orajora/rajni: Path to complete data

/home/rajni/projects/def-orajora/rajni/454: 454 rawreads, filtering using solexa

/home/rajni/projects/def-orajora/rajni/illumina: Illumina raw reads. Trinity filtered reads (40, 50 and 64bp reads) and Transabyss (64bp filtered reads were assembled using TRransAbyss)

/home/rajni/projects/def-orajora/rajni/RedSpruce: Trinity assembly steps (including the config file which is edited according to the single end reads).

/home/rajni/projects/def-orajora/rajni/SEPT22_RedSpruceTRANS_Data_scratch: May sub-directories are present in this directories: This data was copied from the scratch to projects.

Sub-directory 454: 
(/home/rajni/projects/def-orajora/rajni/SEPT22_RedSpruceTRANS_Data_scratch/454): Original raw reads and Trinity assembly (using Galaxy).

Sub-directory annotation and annotations_complete: In the annotations_complete all annotations are copied and in annotation few are missing. SO PLEASE USE annotations_complete directory.

(/home/rajni/projects/def-orajora/rajni/SEPT22_RedSpruceTRANS_Data_scratch/annotations): GO, KEGG, NT, Swissprot, TAIR and TFdb.

Sub-directory CD_hitclust_com_Trinity: (/home/rajni/projects/def-orajora/rajni/SEPT22_RedSpruceTRANS_Data_scratch/CD_hitclust_com_Trinity).

Sub-directory illumina: (/home/rajni/projects/def-orajora/rajni/SEPT22_RedSpruceTRANS_Data_scratch/illumina/originalfiles/filteringtrimmomatic): Filtered illumina reads which were used for the final assembly. There are two more subdirectories present in this directory (Trinity and TransAbyss). In these directories the Illumina reads were assembled using TRINITY and TRANABYSS assembler.

Sub-directory quickmergeassembly:
/home/rajni/projects/def-orajora/rajni/SEPT22_RedSpruceTRANS_Data_scratch/quickmergeassembly): The assemblies were merged but the BUSCO results were not of high quality. So we decided not to use quickmerge assemblies from two different assemblers

Sub-directory SNPs: 
(/home/rajni/projects/def-orajora/rajni/SEPT22_RedSpruceTRANS_Data_scratch/SNPs): SNPs were detected.

Sub-directory 454_35bp_analysis: 
(/home/rajni/projects/def-orajora/rajni/SEPT22_RedSpruceTRANS_Data_scratch/454_35bp_analysis): The assembly was also performed with reads with 35bp cutoff.

Sub-directory DGE: 
(/home/rajni/projects/def-orajora/rajni/SEPT22_RedSpruceTRANS_Data_scratch/DGE): The DGE analysis results.

Sub-directory RAY_combined:
(/home/rajni/projects/def-orajora/rajni/SEPT22_RedSpruceTRANS_Data_scratch/RAY_combined): All six files (Illumina + 454) were assembled using RAY.

Sub-directory Trinity_combined:
(/home/rajni/projects/def-orajora/rajni/SEPT22_RedSpruceTRANS_Data_scratch/Trinity_combined): All six files (Illumina + 454) were assembled using Trinity. In this directory, another directory is also present BLAST (PLEASE IGNORE THIS DIRECTORY). We did the test run in this directory and have used the Blast with DIAMOND.

/home/rajni/projects/def-orajora/rajni/comparativeCHL: The comparative analysis was done for 12 chloroplast genome of Picea species. These genomes were downloaded from NCBI. If anyone wants to use these genomes, I request them to download from NCBI as the updated information and genomes sequences are being updated and improved daily.  

/home/rajni/projects/def-orajora/rajni/software: This directory contains different softwares:
build_faststructure.txt, busco_downloads, env-ntjoin, fastStructure, KAAS, kofamscan, NGSQCToolkit,  ntJoin,  quickmerge,  SOAPdenovo2,  TPMCalculator.

/home/rajni/projects/def-orajora/rajni/DataBases: UniProt Database (needs to be updated).

/home/rajni/projects/def-orajora/rajni/MIRA: MIRA assembler

/home/rajni/projects/def-orajora/rajni/rajniredspruceanalysis: (Repeat assembly, please ignore, I am not deleting this directory)

/home/rajni/projects/def-orajora/rajni/RedSprucetranscriptome: (Two directories: Trinity and Illumina). In Illumina, copy of original raw reads and in Trinity (assembly copy)

Chloroplast genome analysis
/home/rajni/projects/def-orajora/rajni/MAYanalysis: Soapdenovo contigs mapped to reference genome and sealer details.

/home/rajni/projects/def-orajora/rajni/MergingRedBlack: Scaffolds of Redspruce were created using Blackspruce as reference.

/home/rajni/projects/def-orajora/rajni/NCBI_Validation: The genes of Red spruce were validated using the NCBI tool.

/home/rajni/projects/def-orajora/rajni/scripts: All the details of mVISTA is present in this directory.

/home/rajni/projects/def-orajora/rajni/Merged_assembly: Two assemblies were merged but the assembly results showed mis-assembly of few contigs so we decided not to use it.




Data on Narval
Red spruce transcriptome data

/home/rajni/projects/def-orajora/rajni: Path to projects
/home/rajni/projects/def-orajora/rajni/BlackSpruce/Originaldata: Black spruce sequence data of 4 samples (5,6,7,8) only transferred by me.

/home/rajni/projects/def-orajora/rajni: TWO DATABASES: diamond_nr_taxid.dmnd.gz and nr.gz

/home/rajni/projects/def-orajora/rajni/SOAPdenovo-Trans: SOAP denovo tool for transcriptome assembly.

/home/rajni/projects/def-orajora/rajni/software: 
interproscan-5.57-90.0  interproscan-5.57-90.0-64-bit.tar.gz  jvarkit  muscle-5.1  pftools3
Red Spruce Chloroplast Genome

/home/rajni/projects/def-orajora/rajni/comparativeCHL: Genomes of Picea species for comparative analysis


